### PR TITLE
projection fix

### LINKED ISSUE #289: Projection for the n-dimensional systems, where n>2 does not work in hybrid setting

### PR DESCRIPTION
**Pull request by kostakoida**
Closes #289.

### CODE REVIEW

File: src/ReachSets/project_reach.jl [hunk 1]
```diff
@@ -120,13 +125,12 @@ This array contains the collection of reach sets in 2D.
...
 in the sequence of 2D sets `Rsets`.
 """
-function project_reach(
+function project_2d_reach(
         Rsets::Vector{<:ReachSet{<:LazySets.LazySet{numeric_type}}},
         vars::Vector{Int64},
         n::Int64,
         options::Options)::Vector{<:ReachSet} where {numeric_type<:Real}
-    # parse input
-    assert(length(vars) == 2)
```
Comment (schillic) [inline src/ReachSets/project_reach.jl]: Why remove that?

File: src/ReachSets/project_reach.jl [hunk 2]
```diff
@@ -17,12 +17,17 @@ The `vars` argument is required even if the optional argument
...
-        Rsets::Vector{<:ReachSet{<:LazySets.CartesianProductArray{numeric_type}}},
+        Rsets::Vector{<:ReachSet{<:LazySets.LazySet{numeric_type}}},
         vars::Vector{Int64},
         n::Int64,
         options::Options)::Vector{<:ReachSet} where {numeric_type<:Real}
+
     # parse input
     assert(length(vars) == 2)
+    if n == 2
+        project_2d_reach(Rsets, vars, n, options)
```
Comment (schillic) [inline src/ReachSets/project_reach.jl]: You have to `return` here :wink:

File: src/ReachSets/project_reach.jl (hunk 1 above)
Comment (kostakoida) [reply src/ReachSets/project_reach.jl]: Why do we need it 2 times? We already checked previously

Comment (schillic) [reply src/ReachSets/project_reach.jl]: Oh, right!

**Pull request merged**